contrib/src/templates/mod.rs

```diff
@@ -200,19 +226,67 @@ fn discover_templates() -> HashMap<String, TemplateInfo> {
         let mut glob_path: PathBuf = [&*TEMPLATE_DIR, "**", "*"].iter().collect();
         glob_path.set_extension(ext);
         for path in glob(glob_path.to_str().unwrap()).unwrap().filter_map(Result::ok) {
-            let rel_path = path.strip_prefix(&*TEMPLATE_DIR).unwrap().to_path_buf();
-            let path_no_ext = remove_extension(&rel_path);
-            let data_type = path_no_ext.extension();
-            let name = remove_extension(&path_no_ext);
-            templates.insert(name.to_string_lossy().into_owned(), TemplateInfo {
-                full_path: path.to_path_buf(),
-                path: rel_path,
-                extension: path.extension().unwrap().to_string_lossy().into_owned(),
-                data_type: data_type.map(|d| d.to_string_lossy().into_owned())
-            });
+            let (rel_path, name, data_type) = split_path(&path);
+            templates.insert(name, TemplateInfo {
```

Comment:
The indentation of the fields for the `TemplateInfo`. I'm not sure what the canonical thing is, though. If `rustfmt` did this and the rustc source does it this way too, then this is okay.